packages/api/src/external/ehr/elation/command/get-patients-from-appointments.ts

```diff
@@ -0,0 +1,249 @@
+        errorCount: getPatientIdOrFailErrors.length,
+        errors: getPatientIdOrFailErrors.join(","),
+        context: "elation.get-patients-from-appointments",
+      },
+    });
+  }
+}
+
+async function getAppointmentsByPractice({
+  cxId,
+  practiceId,
+  patientAppointments,
+  fromDate,
+  toDate,
+  errors,
+  log,
+}: {
+  cxId: string;
+  practiceId: string;
+  patientAppointments: PatientAppointment[];
+  fromDate: Date;
+  toDate: Date;
+  errors: string[];
+  log: typeof console.log;
+}): Promise<void> {
+  try {
+    const { clientKey, clientSecret } = await getElationClientKeyAndSecret({
+      cxId,
+      practiceId,
+    });
+    const api = await ElationApi.create({
+      practiceId,
+      environment: elationEnvironment as ElationEnv,
+      clientKey,
+      clientSecret,
+    });
+    const appointments = await api.getAppointments({
+      cxId,
+      fromDate,
+      toDate,
+    });
+    patientAppointments.push(
+      ...appointments.map(appointment => {
+        return {
+          cxId,
+          elationPracticeId: practiceId,
+          elationPatientId: appointment.patient,
+        };
+      })
+    );
+  } catch (error) {
+    const cause = `Cause: ${errorToString(error)}`;
+    const details = `cxId ${cxId} practiceId ${practiceId}.`;
+    const msg = "Failed to get appointments.";
+    log(`${details} ${msg} ${cause}`);
+    errors.push(`${msg} ${details} ${cause}`);
+  }
+}
+
+async function getPatientIdOrFailByPractice({
+  cxId,
+  practiceId,
+  patientAppointmentsUnique,
+  errors,
+  log,
+}: {
+  cxId: string;
+  practiceId: string;
+  patientAppointmentsUnique: PatientAppointment[];
+  errors: string[];
+  log: typeof console.log;
+}) {
+  const { clientKey, clientSecret } = await getElationClientKeyAndSecret({
+    cxId,
+    practiceId,
+  });
+  const api = await ElationApi.create({
+    practiceId,
+    environment: elationEnvironment as ElationEnv,
+    clientKey,
+    clientSecret,
+  });
+  const getPatientIdOrFaiLArgs = patientAppointmentsUnique.map(appointment => {
+    return {
+      api,
+      cxId: appointment.cxId,
+      elationPracticeId: appointment.elationPracticeId,
+      elationPatientId: appointment.elationPatientId,
+      triggerDq: true,
+      errors,
+      log,
+    };
+  });
+
+  await executeAsynchronously(getPatientIdOrFaiLArgs, getPatientIdOrFail, {
+    numberOfParallelExecutions: parallelPatients,
+    delay: delayBetweenPracticeBatches.asMilliseconds(),
+  });
+}
+
+async function getPatientIdOrFail({
+  api,
+  cxId,
+  elationPracticeId,
+  elationPatientId,
+  triggerDq,
+  errors,
+  log,
+}: {
+  api: ElationApi;
+  cxId: string;
+  elationPracticeId: string;
+  elationPatientId: string;
+  triggerDq: boolean;
+  errors: string[];
+  log: typeof console.log;
+}): Promise<void> {
+  try {
+    await singleGetPatientIdOrFail({
+      api,
+      cxId,
+      elationPracticeId,
+      elationPatientId,
+      triggerDq,
+    });
+  } catch (error) {
```

Comment:
What's the goal of this function?

Reply:
Renamed, but it's the function that tries to map the elation patient into metriport. Up until this point the command is simply fetching patient IDs from appointments (but not performing any metriport lookup).

Reply:
No longer relevant.